src/utils/dom-writer.js

```diff
@@ -178,31 +178,33 @@ export class DomWriterStreamer {
 
   /** @private */
   merge_() {
-    this.mergeScheduled_ = false;
-
     // Body has been newly parsed.
     if (!this.targetBody_ && this.parser_.body) {
       this.targetBody_ = this.onBody_(this.parser_);
     }
 
-    // Merge body children.
     if (this.targetBody_) {
-      const inputBody = dev().assertElement(this.parser_.body);
-      const targetBody = devAssert(this.targetBody_);
-      let transferCount = 0;
-      removeNoScriptElements(inputBody);
-      while (inputBody.firstChild) {
-        transferCount++;
-        targetBody.appendChild(inputBody.firstChild);
-      }
-      if (transferCount > 0) {
-        this.onBodyChunk_();
-      }
-    }
-
-    // EOF.
-    if (this.eof_) {
-      this.onEnd_();
+      this.targetBody_.then(targetBody => {
+        this.mergeScheduled_ = false;
+
+        // Merge body children.
+        const inputBody = dev().assertElement(this.parser_.body);
+        devAssert(targetBody);
+        let transferCount = 0;
+        removeNoScriptElements(inputBody);
+        while (inputBody.firstChild) {
+          transferCount++;
+          targetBody.appendChild(inputBody.firstChild);
+        }
+        if (transferCount > 0) {
+          this.onBodyChunk_();
+        }
+
+        // EOF.
+        if (this.eof_) {
+          this.onEnd_();
```

Comment:
Wanted to clarify that the intention here is for this to be called after DOM has been transferred? The jsdoc says `Sets the callback that will be called when the DOM has been fully constructed` but I wasn't sure

Reply:
eh I guess there are two cases, after transfer in normal case, or if `writer.close()` is called and there is no `parser.body` tried to handle both